tls/s2n_shutdown.c

```diff
@@ -116,21 +123,25 @@ int s2n_shutdown(struct s2n_connection *conn, s2n_blocked_status *blocked)
         return S2N_SUCCESS;
     }
 
+    /* Skip any leftover plaintext not consumed by the application.
+     * Partial or encrypted records should not be wiped-- we need to finish
+     * processing them or report an error.
+     */
+    if (conn->in_status == PLAINTEXT) {
+        POSIX_GUARD_RESULT(s2n_record_wipe(conn));
+    }
```

Comment:
I think you're right because of [this check](https://github.com/aws/s2n-tls/blob/main/tls/s2n_recv.c#L65-L69), and the new tests agree. That must have been why I removed this is the first place :P I'll remove it again.